### PR TITLE
Autoformat with black

### PR DESCRIPTION
**Pull request by Carreau**
Prototype, 

see https://twitter.com/Mbussonn/status/1128841958711889920

### CODE REVIEW

**Comment by stonebig** [comment]
I replaced shortcuts.py per this one, but nothing happens...
Black is there... did I miss something ?
image

**Comment by stonebig** [comment]
ah, .. it works only on ipython "DOS" console ?

File: IPython/terminal/shortcuts.py [hunk 1]
```diff
@@ -87,10 +87,23 @@ def create_ipython_shortcuts(shell):
 
 
 def newline_or_execute_outer(shell):
+
+    import black
     def newline_or_execute(event):
         """When the user presses return, insert a newline or execute the code."""
         b = event.current_buffer
         d = b.document
+        def try_reformat():
```
Comment (LucianaMarques) [inline IPython/terminal/shortcuts.py]: I think it's a little bit confusing to define a function inside of another function. Could you please tell me whether there's a specific reason for this?

**Review by LucianaMarques — COMMENTED** [review]
I'm not getting where you are checking whether black is installed, could you please point it out for me? :) Thanks!

File: IPython/terminal/shortcuts.py (hunk 1 above)
Comment (Carreau) [reply IPython/terminal/shortcuts.py]: For both of your questions the answer is : "there is no reason beyond pure lazyness", I was prototyping and looking at "can this be done"; you are right that a proper implementation should move the import to the top of the file and define the `try_reformat()` function outside, passing in the `b` buffer.

Comment (LucianaMarques) [reply IPython/terminal/shortcuts.py]: Ok, fair enough :)

How do you check whether black is activated?

**Comment by Carreau** [comment]
I've pushed some update, including some documentation; and will try to get that in a a preview for 7.10 at end of month.

**Pull request merged**